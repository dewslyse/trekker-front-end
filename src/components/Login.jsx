/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useEffect, useRef, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Navigate } from 'react-router-dom';
import { loginUser } from '../store/actions/userActions';

const Login = () => {
  const userRef = useRef();
  const dispatch = useDispatch();
  const isLoggedIn = useSelector((state) => state.user.isLoggedIn);

  const [user, setUser] = useState({
    username: '',
    password: '',
  });

  useEffect(() => {
    userRef.current.focus();
  }, []);

  const onInputChange = (e) => {
    setUser({
      ...user,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(loginUser(user));
    setUser({
      username: '',
      password: '',
    });
  };

  return (
    <>
      {isLoggedIn && (<Navigate to="/destinations/1" replace />)}
      <section>
        <h1>Sign In</h1>
        <form onSubmit={handleSubmit}>
          <label htmlFor="username">Username</label>
          <input
            type="text"
            id="username"
            ref={userRef}
            onChange={onInputChange}
            value={user.username}
            name="username"
            required
          />
          <label htmlFor="password">Password</label>
          <input
            type="password"
            id="password"
            onChange={onInputChange}
            value={user.password}
            name="password"
            required
          />
          <button type="submit">Sign In</button>
        </form>
      </section>
    </>
  );
};

export default Login;
