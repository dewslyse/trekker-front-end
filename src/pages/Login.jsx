/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useEffect, useRef, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, Navigate } from 'react-router-dom';
import { loginUser } from '../store/actions/userActions';
import { hideNotification } from '../store/reducers/uiReducers';
import trekker from '../trekker.png';
import styles from './Login.module.scss';

const Login = () => {
  const userRef = useRef();
  const dispatch = useDispatch();
  const isLoggedIn = useSelector((state) => state.user.isLoggedIn);
  const [loading, setLoading] = useState(false);

  const [user, setUser] = useState({
    username: '',
    password: '',
  });

  const {
    loginContainer, loginForm,
  } = styles;

  useEffect(() => {
    userRef.current.focus();
  }, []);

  const onInputChange = (e) => {
    setUser({
      ...user,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    setLoading(true);
    dispatch(loginUser(user))
      .unwrap()
      .then(() => {
        localStorage.setItem('LOGGED_IN', user.username);
        setTimeout(() => dispatch(hideNotification()), 2000);
      })
      .catch(() => { setLoading(false); });
    setUser({
      username: '',
      password: '',
    });
  };

  return (
    <>
      {isLoggedIn && (<Navigate to="/destinations" replace />)}
      <section className={loginContainer}>
        <div className={loginForm}>
          <img src={trekker} alt="trekker logo" />
          <form onSubmit={handleSubmit} method="">
            <label htmlFor="username">Username</label>
            <input
              type="text"
              id="username"
              ref={userRef}
              onChange={onInputChange}
              value={user.username}
              name="username"
              required
            />
            <label htmlFor="password">Password</label>
            <input
              type="password"
              id="password"
              onChange={onInputChange}
              value={user.password}
              name="password"
              required
            />
            <button type="submit">Sign In</button>
            <Link to="/register">Register</Link>
          </form>
        </div>
      </section>
      {loading && <p>Loading...</p>}
    </>
  );
};

export default Login;
