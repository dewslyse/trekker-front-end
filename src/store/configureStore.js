import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import destinationReducer from './reducers/destinationReducer';
import reservationReducer from './reducers/reservationReducer';
import uiReducer from './reducers/uiReducers';
import userReducer from './reducers/userReducer';

const store = configureStore({
  reducer: {
    destinations: destinationReducer,
    reservations: reservationReducer,
    user: userReducer,
    ui: uiReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  devTools: process.env.NODE_ENV !== 'production',
});

export default store;
