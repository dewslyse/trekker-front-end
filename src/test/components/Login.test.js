import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Login from '../../pages/Login';
import store from '../../store/configureStore';

describe('Login to app', () => {
  it('renders correctly', () => {
    const tree = render(<Provider store={store}><Router><Login /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});