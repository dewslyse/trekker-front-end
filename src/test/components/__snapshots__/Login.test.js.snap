// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login to app renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="loginContainer"
      >
        <div
          class="loginForm"
        >
          <img
            alt="trekker logo"
            src="trekker.png"
          />
          <form
            method=""
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              id="username"
              name="username"
              required=""
              type="text"
              value=""
            />
            <label
              for="password"
            >
              Password
            </label>
            <input
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
            <button
              type="submit"
            >
              Sign In
            </button>
            <a
              href="/register"
            >
              Register
            </a>
          </form>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="loginContainer"
    >
      <div
        class="loginForm"
      >
        <img
          alt="trekker logo"
          src="trekker.png"
        />
        <form
          method=""
        >
          <label
            for="username"
          >
            Username
          </label>
          <input
            id="username"
            name="username"
            required=""
            type="text"
            value=""
          />
          <label
            for="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
          <button
            type="submit"
          >
            Sign In
          </button>
          <a
            href="/register"
          >
            Register
          </a>
        </form>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
